api/src/main/java/org/openmrs/api/ConceptService.java

```diff
@@ -1922,5 +1922,57 @@ public Drug getDrugByMapping(String code, ConceptSource conceptSource,
 	 */
 	@Authorized(PrivilegeConstants.GET_CONCEPTS)
 	boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType);
-	
+	/**
+	 * Get Concept by id or name convenience method
+	 * 
+	 * @param idOrMap
+	 * @return the found Concept
+	 * @throws APIException
+	 */
+	@Authorized(PrivilegeConstants.GET_CONCEPTS)
+	public static Concept getConceptInfo(String idOrMap) throws APIException {
+
+		Concept cpt = null;
+
+		if (idOrMap != null) {
+
+			idOrMap = idOrMap.trim();
+
+			//if input contains at least one period handle recursively as a code constant
+			if (idOrMap.contains(".")) {
+				return getConcept(evaluateStaticConstant(idOrMap));
+			}
+
+			// handle  mapping id: xyz:ht
+			int index = id.indexOf(":");
+			if (index != -1) {
+				cpt = Context.getService(HtmlFormEntryService.class).getConceptByMapping(idOrMap);
+				if (cpt != null) {
+					return cpt;
+				}
+			}
+
+			// handle uuid id: "a3e1302b-74bf-11df-9768-17cfc9833272", if the id matches a uuid format
+			if (isValidUuidFormat(idOrMap)) {
+				cpt = Context.getConceptService().getConceptByUuid(idOrMap);
+				if (cpt != null) {
+					return cpt;
+				}
+			}
+
+			//finally, see if this is a parseable int; if so, try looking up concept by id
```

Comment:
it's always better not to mix the comment with code lines. In fact try, not to write comments unless it is really needed. Your code should express what it does. You can choose nice variable names and method names to achieve that. :-)